File: BOJ/python3/11047.py
'''
동전 0
https://www.acmicpc.net/problem/11047
'''
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력

coins = []

N, K = map(int, sys.stdin.readline().split())
for _ in range(N):
    coin = int(sys.stdin.readline())
    coins.append(coin)

coins.sort(reverse=True)
cnt = 0
for coin in coins:
    if K // coin != 0:
        cnt += K // coin
        K = K % coin

    if K == 0:
        print(cnt)
        break


# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/11047.js
/*
동전 0
https://boj.kr/11047
 */
const start = new Date().getTime();

const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .split("\n");

const [N, K] = input[0].split(" ").map(Number);
const A = input.slice(1).map(Number);

let ans = 0;
let r = K;
for (let i = N - 1; i >= 0; i--) {
  if (r === 0) break;
  if (A[i] > r) continue;

  ans += Math.floor(r / A[i]);
  r %= A[i];
}

console.log(ans);

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
